content/posts/blog/2020-10-30-consuming-public-content.md

```diff
@@ -0,0 +1,108 @@
+
+* **docker run nginx** (or other public images)?
+
+* **helm install stable/cert-manager** (or other helm charts that pull images from public registries)?
+
+If so, this article applies to you.
+
+## Consuming Any Public Content
+
+The overarching public content distribution problem isn’t limited to who should bear the cost for public content, but rather also encompasses who bears the responsibility of assuring the content is accessible and secure for your environment, 100% of the time. The recent Docker TOS changes encourage us to ask a broader set of questions that, as a community, we can address. The problem isn’t limited to production container images but extends to all package manager content (debs, rpms, rubygems, node modules, etc).
+
+While the Docker Terms of Service (TOS) update directly imposes throttling on frequent non-authenticated and free account content pulls, which may impact critical workloads, it also raises the question of how and when public content should impact critical workloads.
+
+## Single Points of Failure
+
+Every application architect asks, where are the single points of failure and how can we mitigate them? As an example, in the top half of the below image, we see a DNS outage, indicated in red, that impacts regions a, c & d across all cloud providers. The DNS single point of failure impacted millions of users across multiple regions and multiple clouds. The lower half represents the same workloads, across all regions, where each region maintains a copy of the content they depend upon. The public content is still consumed, however, each customer maintains a copy of the content they depend upon within a registry they operate, thus avoiding the single point of failure.
+
+![image alt text](/img/blog/2020-10-30-consuming-public-content/consuming1.png)
+
+## Cost Doesn’t Mitigate Risk
+
+If Docker, Quay, GitHub, or other public registries undertake the financial burden for serving content to the world, who bears the burden of reliability for all the connections between critical workloads and the public registry? Even if all inter-connections were 100% reliable 100% of the time and the risk of bad actors inserting vulnerabilities were not possible, who bears the impact of well-intentioned security updates causing an outage to critical workloads? As an example, when maintainers of the "[node" image updated symbolic references to yarn](https://github.com/nodejs/docker-node/issues/649), many consumers of the image were broken. While many focus on the underlying issue of using specific digests and tags to reference a desired version of a container image, this example highlighted the tension between getting security updates and the stability of content, for *your environment*. The question of access to public content isn’t just a matter of making public content cheap and reliably accessible, it’s a matter of creating gated buffers of the public content we depend upon.
```

Comment:
```suggestion
If Docker, Quay, GitHub, or other public registries undertake the financial burden for serving content to the world, who bears the burden of reliability for all the connections between critical workloads and the public registry? Even if all inter-connections were 100% reliable 100% of the time and the risk of bad actors inserting vulnerabilities were not possible, who bears the impact of well-intentioned security updates causing an outage to critical workloads? As an example, when maintainers of the ["node" image updated symbolic references to yarn](https://github.com/nodejs/docker-node/issues/649), many consumers of the image were broken. While many focus on the underlying issue of using specific digests and tags to reference a desired version of a container image, this example highlighted the tension between getting security updates and the stability of content, for *your environment*. The question of access to public content isn’t just a matter of making public content cheap and reliably accessible, it’s a matter of creating gated buffers of the public content we depend upon.
```

Reply:
Can't see the difference...?

Reply:
Not sure why GitHub isn't highlighting it like it normally does, but at least `"[node" ...` vs `["node" ...` is changed in here